lib/puppet/provider/openldap_database/olc.rb

```diff
@@ -119,6 +119,7 @@ def create
     t = Tempfile.new('openldap_database')
     t << "dn: olcDatabase=#{resource[:backend]},cn=config\n"
     t << "changetype: add\n"
+    t << "objectClass: olcDatabaseConfig\n"
```

Comment:
Well you're right, there is a bug in `self.instances`, I think that you should fix it there (in `self.instances`).